src/components/appUpdate/useUpdateDialog.native.js

```diff
@@ -52,13 +54,29 @@ export default () => {
         })
       }
 
+      // statusCode = 0 - The app is up-to-date with the CodePush server.
+      // statusCode = 2 The app had an optional update which the end user chose to ignore. (This is only applicable when the updateDialog is used)
+      // statusCode = 1 - The update has been installed and will be run either immediately after the syncStatusChangedCallback function returns or the next time the app resumes/restarts, depending on the InstallMode specified in SyncOptions.
+      // statusCode = 4 - There is an ongoing sync operation running which prevents the current call from being executed.
+
       if (suggestCodePushUpdate && !hasNewVersion) {
         await codePush
-          .sync({
-            updateDialog: false,
-            installMode: InstallMode.IMMEDIATE,
-            deploymentKey: codePushDeploymentKey,
-          })
+          .sync(
+            {
+              updateDialog: false,
+              installMode: InstallMode.IMMEDIATE,
+              deploymentKey: codePushDeploymentKey,
+            },
+            status => {
```

Comment:
1. made it simpler e.g. `status === 0`, no need to `switch`
2. do not use local storage, just use TogglesContext and write there directly

Reply:
applied suggestion

Reply:
@L03TJ3 just `setHasSyncedCodePush(status === 0)`